### ISSUE TITLE
Update all files to new header

### ISSUE THREAD

**Issue by FBumann** [flixOpt/flixOpt#84]

**Comment by FBumann** [comment]
Wie sollen wir den header der Dateien gestalten?
Aktuell steht überall:

```
# -*- coding: utf-8 -*-
"""
Created on Wed Dec 16 12:40:23 2020
developed by Felix Panitz* and Peter Stange*
* at Chair of Building Energy Systems and Heat Supply, Technische Universität Dresden
"""
```

**Comment by FBumann** [comment]
# Modern Header Practices for Python Files

## Introduction
When structuring Python files, it's important to follow modern conventions to ensure clarity, maintainability, and compatibility with professional standards. This guide outlines best practices for file headers, metadata, and documentation.

---

## 1. File-Level Docstring
Each Python file should include a concise docstring at the top, explaining its purpose and functionality. Avoid including personal details or timestamps directly in the file.

**Example:**
```
"""
Data processing utilities for energy system models.

This module includes methods for aggregating time series data and
generating visualizations. Part of the flixOpt package.
"""
```

## 2. Metadata in __init__.py or __version__.py
Project-wide metadata, such as authorship and licensing, should be stored in a top-level file like `__init__.py` or `__version__.py`. This ensures consistency and makes metadata easily accessible.

```
__author__ = "Felix Panitz, Peter Stange"
__copyright__ = "Technische Universität Dresden"
__license__ = "MIT"
__version__ = "1.0.0"
```
This approach avoids cluttering individual files with repetitive information while maintaining easy access to metadata.

## 3. Encoding declarations
Encoding declarations like `# -*- coding: utf-8 -*-` are a remnant of Python 2. In Python 3, UTF-8 is the default encoding, so these lines are unnecessary unless compatibility with legacy systems is required.


# setup.py
Alternatively to storing version and author information in the package files itself, using setup.py is best practice

**Comment by FBumann** [comment]
I would suggest we remove the whole author and time stamp stuff, as well as the utf8 stuff, and put everything in the setup.py (what we already have mostly).
Then we can use the file headers for its common purpose: information about the content of the file and its purpose in the package

**Comment by baumbude** [comment]
Können wir gerne so machen. Allerdings ggf. besser in eine license.txt ablegen.

**Comment by FBumann** [comment]
Gut, licence.txt ist gesetzt!
Der Rest kommt in setup.py.
Ich würde so oder so bald auf pyproject.toml updaten. Da ist dann alles auch besser strukturiert drin
[https://packaging.python.org/en/latest/tutorials/packaging-projects/](https://packaging.python.org/en/latest/tutorials/packaging-projects/)
[https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-pyproject-toml](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-pyproject-toml)
Denke das wird aber eher 1.0.1 oder so

**Comment by FBumann** [comment]
Alles auf englisch.
Python Version 3.9 weg

**Comment by FBumann** [comment]
Closed by #90
